src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -114,9 +115,33 @@ module Make (Inputs : Inputs_intf) :
 
     let create () = {snark_pool_refcount= Snark_pool_refcount.create ()}
 
-    let handle_diff t diff =
-      let use handler field = handler (Field.get field t) diff in
-      Fields.iter ~snark_pool_refcount:(use Snark_pool_refcount.handle_diff)
+    type writers =
+      { snark_pool:
+          ( Snark_pool_refcount.view
+          , Strict_pipe.crash Strict_pipe.buffered
+          , unit )
+          Strict_pipe.Writer.t }
+
+    type readers = {snark_pool: unit Strict_pipe.Reader.t}
+
+    let make_pipes () : readers * writers =
+      let snark_reader, snark_writer =
+        Strict_pipe.create
+        @@ Strict_pipe.Buffered (`Capacity 3, `Overflow Strict_pipe.Crash)
+      in
+      ({snark_pool= snark_reader}, {snark_pool= snark_writer})
+
+    let close_pipes ({snark_pool} : writers) =
+      Strict_pipe.Writer.close snark_pool
+
+    let mb_write_to_pipe diff ext_t handle pipe =
+      match handle ext_t diff with
+      | None -> ()
+      | Some new_view -> Strict_pipe.Writer.write pipe new_view
+
+    let handle_diff t (pipes : writers) diff =
+      mb_write_to_pipe diff t.snark_pool_refcount
+        Snark_pool_refcount.handle_diff pipes.snark_pool
```

Comment:
If we wanted to keep the Fields.iter , we could maybe just extend the little `use` function to match on the return value of handler and write to the pipe if necessary? Not sure what `mb` stands for.

Reply:
It stands for maybe. A Haskellism. I can probably get the fields thing to work. I spent a little time thinking about how to make a functor that would turn the list of extension modules into a module that has the records and functions applied to them. I decided if that's possible it's probably super ugly. But the fields thing is simple.

Reply:
`[@@deriving fields]` is great!

Reply:
nit: (for the mb_* function) This maybe should use the `Option.*` combinators instead of being a separate function (`Option.map (* ... *) |> Option.value ~default:Deferred.return`)